url-shortener/backend/src/Backend.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeFamilies #-}
+module Backend where
+
+import Common.Route
+import Control.Concurrent
+import Control.Monad.IO.Class
+import qualified Data.Aeson as A
+import Data.ByteString.Lazy
+import Data.Text (Text)
+import Data.Word
+import Database.Id.Class
+import Data.Pool
+import Database.PostgreSQL.Simple
+import Gargoyle.PostgreSQL.Connect
+import Prelude hiding (id)
+import Obelisk.Backend
+import Obelisk.Route
+import Snap.Core
+
+maxUrlSize :: Word64
+maxUrlSize = 2000
+
+migration :: Query
+migration = "CREATE TABLE IF NOT EXISTS urls\
+  \ (id SERIAL PRIMARY KEY, url VARCHAR(?) NOT NULL);"
+
+backend :: Backend BackendRoute FrontendRoute
+backend = Backend
+  { _backend_run = \serve -> do
+      withDb "db" $ \pool -> do
+        _ <- withResource pool $ \dbcon -> execute dbcon migration [maxUrlSize]
+        serve $ \case
+
+          BackendRoute_Get_Url :/ id -> do
```

Comment:
Let's not name shadow `id` from Prelude. I know you're `hiding (id)` but that's very un-idiomatic.